### ISSUE TITLE
Feature: configuration option to set focus when dialog opens

### ISSUE THREAD

**Issue by kagemusha** [yapplabs/ember-modal-dialog#70]
for modals with input fields, seems natural to set focus to the first one when it opens.  I propose a configuration option such as:

    {{#modal-dialog
        ...
        focusSelector='#some-input-field'
    }}

If you think this useful, I can submit a PR

**Comment by sandstrom** [comment]
I think you can subclass the modal dialog and solve it using the `didInsertElement` hook.

**Comment by kagemusha** [comment]
@sandstrom yes not too difficult to do, but idea is to save that bit of work

**Comment by jrowlingson** [comment]
This would be a nice addition.

**Comment by toranb** [comment]
Just got a request for this from a client -anyone open to me adding it?

**Comment by toranb** [comment]
For now I just sub classed it but let me know if you land that PR boss :)

**Comment by chrislopresto** [comment]
@toranb Definitely. I think it's a very common use case. Implementing this will also help with key responders that require focus as well. 

We may (or may not) need different strategies for `modal-dialog` (uses ember-wormhole) and `tether-dialog` (uses ember-tether). Feel free to drop me a line in the Ember JS Slack if anything overly unexpected presents itself.

**Comment by jdhines** [comment]
I don't know that you can use `didInsertElement`. I've just been running into an issue on my own modal, and this only works the first time the modal is inserted, not any subsequent times. I've been trying to figure out how Discourse does it and hoping I don't have to rewrite all my modal components.

**Comment by coreypmurphy** [comment]
+1 for this feature
